yt_dlp/__init__.py

```diff
@@ -934,7 +934,7 @@ def _real_main(argv=None):
         if opts.rm_cachedir:
             ydl.cache.remove()
 
-        updater = Updater(ydl)
+        updater = Updater(ydl, opts.update_self if isinstance(opts.update_self, str) else None)
```

Comment:
Why not https://github.com/yt-dlp/yt-dlp/pull/6220#discussion_r1112596797, without the change to helptext

Reply:
We cannot discern between `--update` and `--update-to` otherwise. `--update-to` should always update, `--update` not. We use `None` to automatically pick channel and update only if newer (old behavior) and if you specify a channel or tag it will try and update to that regardless.

Example: channel is `stable` and version is higher. `--update` will not update, `--update-to stable` will.